app/models/resource_identifier.rb

```diff
@@ -0,0 +1,33 @@
+##
+# Returns a unique resource identifier for the current state of a given scanned
+#   resource ID in Fedora.
+class ResourceIdentifier
+  attr_reader :id, :repository
+  def initialize(id, repository = nil)
+    @id = id
+    @repository = repository || SolrRepository
+  end
+
+  # Unique key of the object pointed to by ID's state.
+  def to_s
+    @digest ||= Digest::MD5.hexdigest(timestamp)
```

Comment:
Seems like a unique key should take more into account than just a timestamp. Any reason to not also include the id in the hash?
```suggestion
    @digest ||= Digest::MD5.hexdigest(@id + timestamp)
```